js/src/jsmath.cpp

```diff
@@ -78,6 +79,15 @@ static bool math_function(JSContext* cx, CallArgs& args) {
   // NB: Always stored as a double so the math function can be inlined
   // through MMathFunction.
   double z = F(x);
+
+  // TaintFox
+  if(args[0].isObject() && args[0].toObject().is<NumberObject>()){
```

Comment:
You can save a bit of typing here and use the isTaintedNumber() helper function.

Reply:
Good idea, I did not know that helper function existed.